### PR TITLE
Add Windows to TravisCI (but not enable yet)

### PR DESCRIPTION
**Pull request by Lapenkov**
Implemented the generation of Windows jobs for Travis. In fact, the first commit generates Travis matrix with Windows jobs, but the following disables them due to long runtime.

The number of jobs for Linux and Mac stays unchanged:
```
lapenkov@devvm1846:~/work/jemalloc  travis_windows
▶ cat .travis.yml | grep -E "(\- os|arch:)" | sort | uniq -c
     87       arch: amd64
     11       arch: ppc64le
     92     - os: linux
      8     - os: osx
      
      
lapenkov@devvm1846:~/work/jemalloc  dev
▶ cat .travis.yml | grep -E "(\- os|arch:)" | sort | uniq -c
     87       arch: amd64
     11       arch: ppc64le
     92     - os: linux
      8     - os: osx
```

Other changes:
- Add `CROSS_COMPILE_32BIT` env variable to allow platform-dependent scripts to install necessary dependencies. For linux, it hides `gcc_multilib` installation into `scripts/linux/before_install.sh`.
- Change `language: generic` to `language: minimal`. This changes the Linux image we use, minimal doesn't ship some language runtimes we don't use anyway. Differences are described [here](https://docs.travis-ci.com/user/languages/minimal-and-generic/). This was needed, because Windows VMs wouldn't launch with `language: generic`.
- Do not include empty environment variables into the job specification. I.e. half of the jobs won't have `COMPILER_FLAGS=""` anymore.

### CODE REVIEW

File: scripts/gen_travis.py [hunk 1]
```diff
@@ -108,94 +131,91 @@ def __eq__(self, obj):
...
     + configure_flag_unusuals + malloc_conf_unusuals)
 
 
-gcc_multilib_set = False
-
-
 def get_extra_cflags(os, compiler):
     # We get some spurious errors when -Warray-bounds is enabled.
+    if os == WINDOWS:
+        return ['-fcommon'] if compiler != CL.value else []
```
Comment (interwq) [inline scripts/gen_travis.py:141]: I'm a bit lost on the syntax here; maybe separate lines?

File: scripts/windows/before_install.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    # a batch script that calls "vcvarsall.bat" and prints the environment variables.
+    #
+    # Then, those environment variables are transformed from cmd to bash format and put
+    # into a script $apply_vsenv. If cl.exe needs to be used from bash, one can
+    # 'source $apply_vsenv' and it will apply the environment variables needed for cl.exe
+    # to be located and function correctly.
+    #
+    # At last, a function "mingw_with_msvc_vars" is generated which forwards user input
+    # into a correct mingw (32 or 64) subshell that automatically performs 'source $apply_vsenv',
+    # making it possible for autotools to discover and use cl.exe.
```
Comment (interwq) [inline scripts/windows/before_install.sh:60]: Thanks for the comments.

**Review by interwq — APPROVED** [review]

File: scripts/gen_travis.py [hunk 3]
```diff
@@ -6,35 +6,52 @@
...
+WINDOWS = 'windows'
 
 
 AMD64 = 'amd64'
 ARM64 = 'arm64'
 PPC64LE = 'ppc64le'
 
 
-TRAVIS_TEMPLATE = """
+TRAVIS_TEMPLATE = """\
```
[the remark this answers is not in the document]
Comment (Lapenkov) [reply scripts/gen_travis.py:17]: Yes, without it the first line of `.travis.yml` is empty.

File: scripts/gen_travis.py [hunk 4]
```diff
@@ -6,35 +6,52 @@
...
 # This config file is generated by ./scripts/gen_travis.py.
 # Do not edit by hand.
 
-language: generic
+language: minimal
```
[the remark this answers is not in the document]
Comment (Lapenkov) [reply scripts/gen_travis.py:21]: Sure, will do

File: scripts/gen_travis.py (hunk 1 above)
Comment (Lapenkov) [reply scripts/gen_travis.py:141]: No problem, will make it more readable

**Pull request merged**